README.md
###The React-Native-Button [![Build Status](https://travis-ci.org/xgfe/react-native-buttons.svg?branch=master)](https://travis-ci.org/xgfe/react-native-buttons)  [![Coverage Status](https://coveralls.io/repos/github/xgfe/react-native-buttons/badge.svg?branch=master)](https://coveralls.io/github/xgfe/react-native-buttons?branch=master)

react native button based on pure JavaScript with good expansibility.

### Main

* This component provide 5 theme color for user to chose.
* This component provide 2 type of button for user to chose.
* This component provide the function that user can define their own theme color, we can help them to calculate the disabled color, active color and so on. Also they can define the disabled color and active color by themselves if they like.
* This component provide the ripple effect of the button. And due to lower Android API which do not have this effect, we use highlight as alternative in the consideration of performance.
* This component provide you add as more other components as you can into the button(eg: icon).

### Example
* API >= 21

![image](https://raw.githubusercontent.com/lulutia/images/master/react-native-button/APIhigh.gif)

* API < 21

![image](https://raw.githubusercontent.com/lulutia/images/master/react-native-button/APIlow.gif)

### How to install

```
npm install react-native-buttons --save
```

### Properties

| Prop  | Default  | Type | Description |
| :------------ |:---------------:| :---------------:| :-----|
| type | 'surface' | `string` | Specify the type of the button, you can chose form 'surface' and 'ghost'  |
| size | 'default' | `string` | Specify the size of the button, you can chose from 'small', 'default', 'large'  |
| theme | 'default' | `string | color type` | Specify the theme color, you can chose from 'orange', 'blue', 'red', 'gray', 'default'. Or you can define your own color by use the colort type(eg: 'gba(221,106,167,0.8)', '#BA55D3' and so on) |
| isLoading | false | `bool` | Specify the button is on loading status. In this status, the button is disabled |
| loadingTitle | 'Loading' | `string` | Specify the text of loading status. |
| loadingColor| - | `color type` | Specify the loading color, if you do not specify this color, we can calculate it for you based on the theme color |
| selfStyle | - | `style type` | Specify button's style by yourself |
| disableColor | - | `color type` | Specify the disabled color, if you do not specify this color, we can calculate it for you based on the theme color |
| active | false | `bool` | Specify the active status of the button |
| disabled | false | `bool` | Specify the disable status of the button|
| activeColor | - | `color type` | Specify the active color, if you do not specify this color, we can calculate it for you based on the theme color|

### Method

see the react native document of TouchableWithoutFeedback. All methods supported by TouchableWithoutFeedback can be used here.

### Usage

* you can see and run the example in the app/index.js.

```js
import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View
} from 'react-native';
import {Button} from 'react-native-buttons';

class button extends Component {
  _onPressButton () {
    console.log('onpress');
  }
  render() {
    return (
      <View style={{marginLeft: 20, marginTop: 20}}>
          <Button
            type="surface"
            size="small"
            theme="orange"
            loadingTitle="正在加载"
            isLoading={true}
            onPress={this._onPressButton}>Small</Button>
          <Button
            selfStyle={{marginLeft: 120}}
            type="ghost"
            size="small"
            theme="blue"
            isLoading={true}
            onPress={this._onPressButton}>Default</Button>
      </View>
      );
  }
}

export default button;

```


CHANGELOG.md
<a name="0.0.1"></a>
## 0.0.1 (2016-09-22)


### Bug Fixes

* **button:** to fix the compile of the files begin with react-native in node-modules ([e5588f1](https://github.com/lulutia/react-native-button/commit/e5588f1))


### Features

* **button:** add README ([a4222d8](https://github.com/lulutia/react-native-button/commit/a4222d8))
* **button:** add test ([817435a](https://github.com/lulutia/react-native-button/commit/817435a))
* **button:** add test and changelog ([1c6d106](https://github.com/lulutia/react-native-button/commit/1c6d106))





app/index.js
import React, {Component} from 'react';
import {
  View,
  Text
} from 'react-native';
import {Button} from 'component';

// data

class button extends Component {
  _onPressButton () {
    console.log('onpress');
  }
  render() {
    return (
      <View style={{marginLeft: 20, marginTop: 20}}>
          <Button
            type="surface"
            size="small"
            theme="orange"
            loadingTitle="正在加载"
            isLoading={true}
            onPress={this._onPressButton}>Small</Button>
          <Button
            selfStyle={{marginLeft: 120}}
            type="ghost"
            size="small"
            theme="blue"
            isLoading={true}
            onPress={this._onPressButton}>Default</Button>
             <Button
            selfStyle={{marginTop: 60}}
            theme="#BA55D3"
            loadingTitle="正在加载"
            isLoading={true}
            disableColor="#00C5CD"
            onPress={this._onPressButton}>Default</Button>
             <Button
            selfStyle={{marginLeft: 120, marginTop: 60}}
            theme="#BA55D3"
            loadingTitle="正在加载"
            isLoading={true}
            disableColor="#00C5CD"
            loadingColor="rgba(221,106,167,0.8)"
            onPress={this._onPressButton}>Default</Button>
             <Button
            selfStyle={{marginLeft: 240, marginTop: 60}}
            theme="#BA55D3"
            isLoading={true}
            onPress={this._onPressButton}>Default</Button>
            <Button
            selfStyle={{marginTop: 120}}
            theme="rgba(221,106,167,0.8)"
            onPress={this._onPressButton}>default-btn</Button>
            <Button
            selfStyle={{marginTop: 120, marginLeft: 110}}
            theme="rgba(221,106,167,0.8)"
            onPress={this._onPressButton}><Text style={{marginRight: 20}}>node</Text>string</Button>
            <Button
            selfStyle={{marginTop: 120, marginLeft: 240}}
            theme="rgba(221,106,167,0.8)"
            isLoading={true}
            onPress={this._onPressButton}><Text style={{marginRight: 20}}>test</Text>test</Button>
             <Button
            selfStyle={{marginTop: 180}}
            theme="#00C5CD"
            size="small"
            onPress={this._onPressButton}>small</Button>
            <Button
            selfStyle={{marginTop: 230}}
            theme="#00C5CD"
            size="default"
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 280}}
            theme="#00C5CD"
            size="large"
            onPress={this._onPressButton}>large</Button>
            <Button
            selfStyle={{marginTop: 350}}
            theme="blue"
            size="default"
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 350, marginLeft: 80}}
            theme="red"
            size="default"
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 350, marginLeft: 160}}
            theme="orange"
            size="default"
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 350, marginLeft: 240}}
            theme="gray"
            size="default"
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 400}}
            theme="blue"
            size="default"
            disabled={true}
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 400, marginLeft: 80}}
            theme="red"
            size="default"
            disabled={true}
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 400, marginLeft: 160}}
            theme="orange"
            size="default"
            type="ghost"
            disabled={true}
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 400, marginLeft: 240}}
            theme="gray"
            size="default"
            type="ghost"
            disabled={true}
            onPress={this._onPressButton}>default</Button>
             <Button
            selfStyle={{marginTop: 450}}
            theme="blue"
            size="default"
            active={true}
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 450, marginLeft: 80}}
            theme="red"
            size="default"
            active={true}
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 450, marginLeft: 160}}
            theme="orange"
            size="default"
            active={true}
            type="ghost"
            onPress={this._onPressButton}>default</Button>
            <Button
            selfStyle={{marginTop: 450, marginLeft: 240}}
            theme="gray"
            size="default"
            active={true}
            type="ghost"
            onPress={this._onPressButton}>default</Button>
      </View>
      );
  }
}

export default button;
